src/vsr/replica.zig

```diff
@@ -7977,6 +7963,7 @@ pub fn ReplicaType(
 
             self.grid.free_set.reset();
             self.grid.free_set_checkpoint.reset();
+            self.client_sessions_checkpoint.reset();
```

Comment:
Feels odd that we are not reseting client_sessions here, but rather do it in `client_sessions_open_callback`. 

I think resetting right before decoding makes sense to me from the perspective of defensive programming the decode code doesn't have to worry whether the object was properly reset. But maybe it makes sense to go on an offensive here? Reset the sessions here, so that, between this moment, and the moment we actually decode them, the sessions are in a well-defined reset state? In decoding, we can then _assert_ that the sessions are reset.